### ISSUE TITLE
Metrics investigations for November 7

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#8500]
Two questions:

1. Event counts graph shows drop in events on 2.18 release. Did [we remove events from 2.18](https://github.com/mozilla-mobile/mozilla-vpn-client/compare/v2.17.0...v2.18.0)?
1. Error alerts: `controller` had a jump last week. Why?

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-5843)
┆Reporter: Matt Cleinman

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

1. No change for 2.18, once I dug a little in. The change happened with 2.17’s deployment, not 2.18. 2.17 and 2.18 show similar events per client. 2.16 and 2.17 show a pretty big difference.
  2. Slight increase, yes. Stronger on 2.18, but exists in all of them. Starts on 11/2, and we released a Guardian version then (though I’m not sure if that’s related). I don’t have a satisfying answer here - the biggest jump of `controller` errors is in localsocketcontroller.cpp, but in the 2.16 and 2.17 versions (as well as 2.18).
    1. I’m not quite sure what is happening here, but it dropped after a few days and is close enough to normal that I’m to going to spend more time on the current investigation - though we should keep an eye on it.

I’ve also updated the team on slack. Closing this out.

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

Thanks to encouragement from Matt Lichtenstein , I dug in some more:

I pulled some info out of looker ( https://docs.google.com/spreadsheets/d/10vJAzPf8L32BV3gix3UeMhPplKClYiHC/edit?usp=sharing&ouid=117967310038774988639&rtpof=true&sd=true ) to manipulate it more easily. (Check out the right most pieces of each sheet to see the colorful part, which IMO is the most relevant.)

(Small screenshot) The big drop in events came in 2.17, not 2.18. And there was a giant increase in events in 2.16.

(Large screenshot) I pulled out the most common events, and tracked them across releases, focusing on 2.16, 2.17, and 2.18 (events shown as a % of all events in that release):

  * Websocket events used to be a major component of our events (77%-80% of all events reported), and we removed them in 2.17.
  * How we report connection_health_no_signal and connection_health_unstable changed in 2.16. And in 2.17 and 2.18, they are 68-75% of all events reported.
  * Our absolute count of events stayed relatively the same, because we sorta swapped out websocket events for connection health events…
  * …except in 2.16, where we had both these types of events. The increase in absolute count of events in 2.16 thus makes sense.
  * This attached chart makes it look like controller_state_off, controller_state_on, controller_step, and connectivity_stable took a major drop in 2.16 only. However, the prior bullet point explains this: They were constant, but 2.16 had relatively more events, causing their % to drop.
!Screenshot.png|width=627,height=63!

!Screenshot 2.png|width=827,height=266!
